### ISSUE TITLE
Video Wake Lock not released after video closed

### ISSUE THREAD

**Issue by xwbdwx** [MehediH/Tweeten#231]
Tweeten Windows app is not releasing video wake lock after closing out of a video within the app. Open an elevated command prompt with the command 'powercfg /requests' to show the wake lock:

c:\>powercfg /requests
DISPLAY:
[PROCESS] \Device\HarddiskVolume2\Program Files\WindowsApps\[...]\app\Tweeten.exe
Video Wake Lock

**Comment by RealSkeptic** [comment]
This also happens in MacOS.

I always have to go view->reload to release.

Not sure if it's because of videos or gifs.

**Comment by MehediH** [comment]
Can you check to see if it happens with tweetdeck.twitter.com on the browser?

**Comment by xwbdwx** [comment]
I can confirm that after a video or gif is watched, then subsequently closed on tweetdeck.twitter.com in a browser (chrome latest tested on Win 10 64 bit), video wake lock is released.

During video playback:

> c:\>powercfg /requests
> DISPLAY:
> [PROCESS] \Device\HarddiskVolume2\Program Files\Google\Chrome\Application\chrome.exe
> Video Wake Lock
> 
> SYSTEM:
> [DRIVER] Sound Blaster Recon3Di (HDAUDIO\[])
> An audio stream is currently in use.
> 
> AWAYMODE:
> None.
> 
> EXECUTION:
> [PROCESS] \Device\HarddiskVolume2\Program Files\Google\Chrome\Application\chrome.exe
> Playing audio
> 
> PERFBOOST:
> None.
> 
> ACTIVELOCKSCREEN:
> None.
> 

After video is closed:

> c:\>powercfg /requests
> DISPLAY:
> None.
> 
> SYSTEM:
> None.
> 
> AWAYMODE:
> None.
> 
> EXECUTION:
> None.
> 
> PERFBOOST:
> None.
> 
> ACTIVELOCKSCREEN:
> None.

**Comment by MehediH** [comment]
Thanks! Are you able to confirm if the video lock issue only occurs when GIF Autoplay is enabled, and you click on a GIF to see the full size? Or does it happen for actual videos too?

**Comment by xwbdwx** [comment]
I can confirm it happens with actual videos, too. Click on video within tweet in Tweeten -> video wake lock. Close video (x) -> video wake lock remains.

**Comment by MehediH** [comment]
I am unable to reproduce this, see the following logs:

When playing a video:

```
C:\WINDOWS\system32>powercfg /requests
DISPLAY:
[PROCESS] \Device\HarddiskVolume2\TweetenApp\node_modules\electron\dist\electron.exe
Video Wake Lock
```

After closing video:

```
C:\WINDOWS\system32>powercfg /requests
DISPLAY:
None.
```

**Comment by jetuser** [comment]
I managed to reproduce this on my windows machine, I did not specifically open any videos and was just scrolling my timeline until I saw a wake lock show up, then I scrolled back to the top of my feed and verified the lock remained.

Oddly enough I didn't immediately notice any videos or gifs in the feed at the time the lock showed up.

**Comment by MehediH** [comment]
@jetuser do you have auto-playing gifs turned off? GIFs are technically videos and acquire video locks when they are automatically playing. Can you please check with GIF auto-playing turned off?

**Comment by jetuser** [comment]
You mean the "automatically pause gifs" option? I turned that on and that seemed to prevent the lock acquisition. If that turns out to be the issue then it seems the gif is somehow retaining the lock even though its not directly visible on your feed anymore.

**Comment by MehediH** [comment]
Yes. 

It is possible that the GIFs could be retaining the locks even if they are not visible on the feed, but unfortunately, this would be a TweetDeck issue as the auto-pausing feature is implemented on TweetDeck itself. I will see if there is any way I can prevent that from happening, but I doubt there is one.

**Comment by RealSkeptic** [comment]
I have GIF pause on, **on a Mac**, and scrolling through Tweetdeck has no issues.

However, I can reproduce the video lock when viewing GIFs.

If I click on  a GIF and then pause it, everything is OK.

But:

* Click on a GIF, it starts animating
* Click on it again, it shows the GIF enlarged in the center and plays it, while still playing the original
* Dismiss the GIF
* Pause the original

Result:

```
% pmset -g assertions | grep DisplaySleep                                                             

   PreventUserIdleDisplaySleep    1
   InternalPreventDisplaySleep    1
   pid 80892(Tweeten): [0x001963f500058b09] 00:01:01 NoDisplaySleepAssertion named: "Video Wake Lock" 
   pid 71(powerd): [0x0019639e00108131] 00:00:53 InternalPreventDisplaySleep named: "com.apple.powermanagement.delayDisplayOff"
```

**Comment by MehediH** [comment]
@RealSkeptic thank you, this specific case will be fixed in the next version.

**Comment by jetuser** [comment]
I'm wondering if something similar may be happening on windows as well. I managed to get another stuck lock some time in the recent past and I only just noticed. The gif pause thing is still on.

**Comment by Mika56** [comment]
Hi,
Can confirm that by enabling GIF auto-pause, Tweeten is now releasing the video locks on Windows

**Comment by MehediH** [comment]
Fixed in 5.4.0
